src/pages/settings/PasswordPage.js

```diff
@@ -156,34 +222,25 @@ class PasswordPage extends Component {
                                 autoCompleteType="password"
                                 textContentType="password"
                                 value={this.state.confirmNewPassword}
-                                onChangeText={confirmNewPassword => this.setState({confirmNewPassword})}
-                                onSubmitEditing={this.handleChangePassword}
-                                onBlur={() => this.onBlurConfirmPassword()}
+                                onChangeText={text => this.clearErrorAndSetValue('confirmNewPassword', text, ['confirmPasswordMatch'])}
+                                hasError={this.state.errors.confirmNewPassword || this.state.errors.confirmPasswordMatch}
+                                errorText={this.getErrorText(this.state.errors.confirmNewPassword ? 'confirmNewPassword' : 'confirmPasswordMatch')}
+                                onSubmitEditing={this.validateAndSubmitForm}
                             />
                         </View>
-                        {!this.state.shouldShowPasswordConfirmError && !isEmpty(this.props.account.error) && (
+                        {!_.some(_.values(this.state.errors)) && !_.isEmpty(this.props.account.error) && (
```

Comment:
Inverting `_.some()` is a bit confusing and the `_.values()` isn't necessary. 

Maybe clearer to just do something like....

```js
_.every(this.state.errors, error => !error);
```